IanGRap/Squish
# Improve death effect

Contributors: Coulter, Liam, Reed, everyone else in the room

Goal: Redo the death particle effect so that it looks gooier/juicier and is less computationally expensive.

Files Changed:
- Player Death Particle Revisited prefab: New particle effect drawing some elements from the original, with new 2D droplets and splats used instead of 3D meshes.
- SquishDetector.cs: Rewrote lines that control particle color in order to work with the new 2D materials and the specific shader type they are using.
- ObjectSpawner.cs: Same as in SquishDetector.cs, so that the new effect is also used in the lobby.
- CameraGoo.cs: script now contains a serialized list of different screen splatter patterns, and the specific image used for each death is picked at random.

How to test:
- Kill players in both the lobby and match scenes. There should be a new goo explosion effect that is is the same color as the skin on the respective player. Different screen splatters, selected randomly each time, should appear on death.

Notes:
- In the code blocks that set the particle color, I included a link to the webpage that I used to figure out how to use the Material Property Block (which is necessary because of the specific particle shader type I am using). If you think the code is straightforward enough, I could remove the comment with the url link.
